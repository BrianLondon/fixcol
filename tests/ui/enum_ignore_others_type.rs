use fixcol_derive::ReadFixed;

#[derive(ReadFixed)]
struct MyStruct {
    #[fixed(width = 5)]
    my_int: u16
}

#[derive(ReadFixed)]
#[fixed(ignore_others = 1)]
enum Thing {
    #[fixed(key = "one")]
    Thing1(#[fixed(width = 5)] u16),
    #[fixed(key = "two", embed = true)]
    Thing2(MyStruct),
}

pub fn main() {}
