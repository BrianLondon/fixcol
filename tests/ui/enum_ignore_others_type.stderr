error: fixcol-derive error: Expected true or false for ignore_others.
  --> tests/ui/enum_ignore_others_type.rs:10:25
   |
10 | #[fixed(ignore_others = 1)]
   |                         ^
